### ISSUE TITLE
[3.7.0-rc3] Pages cache not working

### ISSUE THREAD

**Issue by bnomei** [k-next/kirby#4423]
Description: 
calling any page on frontend, no login to panel, trying to populate pages cache.

Setup:
osx valet nginx php 8.1,
fortrabbit nginx php 8.1

Research:
all (most?) requests seem to be flagged to use auth which breaks the pages cache. this line sets it to `true`
https://github.com/getkirby/kirby/blob/a085ec9c78d9a871ec7b9f4428f45d69bcfd7241/src/Http/Request.php#L164

the pages cache checks if it can cache here
https://github.com/getkirby/kirby/blob/6b20fa11843f57cd9a1e611bc9e8e8a91b855156/src/Cms/Page.php#L1067 
`$response()->cache()` seems to be `false` for my regular content page.

that calls the CMS\Responder cache which thinks the request is private because of (wrongly?) set auth flag from above and which returns false resulting in no cache allowed.
https://github.com/getkirby/kirby/blob/a085ec9c78d9a871ec7b9f4428f45d69bcfd7241/src/Cms/Responder.php#L120

**Comment by lukasbestle** [comment]
@bnomei I cannot reproduce this in the Starterkit. I'll try to explain the behavior to make it easier for you to debug this further.

## Why `$request->auth()` calls `$response->usesAuth(true)`

The line is part of the caching refactoring in #4277. You can read a summary of the goals of this refactoring in this comment: https://github.com/getkirby/kirby/issues/3976#issuecomment-1094297863

**Example:** Let's say you have a special navigation snippet for authenticated users. This snippet will output the current username and a few helpful buttons for users who are authenticated with the `Authorization` header (e.g. HTTP basic auth). In this example, the following should happen:

1. Anonymous visitors should get the normal template output without these special admin buttons. This output can be cached and used for any anonymous visitor. But not for authenticated users because you will want to output the buttons for them.
2. Authenticated users will get their specific personalized template output that only they should see. So no caching at all.

**Transferred to Kirby code:** If any code calls `$request->auth()`, this means that some part of your template output depends on the basic fact whether a user is authenticated or not. And if so, it will likely depend on the value of the `Authorization` header (meaning: on the specific user who is authenticated). This means that:

1. "public" responses (header is not set) can be cached, but the cache cannot be used for authenticated users, because they expect to get a personalized response.
2. "private" responses (header is set) can never be cached because of the private data that only this particular user may see.

The call to `$response->usesAuth(true)` marks the response as reliant on the `Authorization` header in general. This enables the "public" caching logic that the response may not be used for authenticated users.

The "private" caching logic is then handled by `$response->isPrivate()`, which checks if the header is actually set.

So there can be three results:

1. Response does not use the header at all (no call to `$response->usesAuth(true)`): Page is cached as normal
2. Response uses the header (call to `$response->usesAuth(true)`) but header is not set (`$response->isPrivate()` returns `false`): Page is cached, but only for unauthenticated visitors (handled by [this code](https://github.com/getkirby/kirby/blob/9247cde801335f81769f7b546cacc87df52e09e8/src/Cms/Page.php#L1041-L1045))
3. `$response->isPrivate()` returns `true`: Page is never cached in the first place

## Why this call to `$response->usesAuth(true)` depends on the `$kirby` object

The `Http\Request` class may be used outside the CMS context. In this case the lazy-loaded `$kirby` object may be `null`. As the Kirby `Responder` object is only created in the CMS context, the "response marking" (call of `$response->usesAuth(true)`) is skipped.

But if a CMS instance exists, the code will behave exactly like if we used `App::instance()->response()->usesAuth(true)`. So you can just ignore that `if` condition mentally.

## Why this call does *not* depend on the fact whether the `Authorization` header is set

As explained above, we need to catch the condition where the template relies on the `Authorization` header but where the header is not actually set. In this condition, the response can be cached, but needs to be marked so that authenticated users will get a personalized response.

The condition can be detected with the combination of **any** call to `$request->auth()` and `$response->isPrivate()` returning `false`.

## TL;DR: What all this means for debugging this issue

- The page cache is **created and used** as long as your site code does not call `$kirby->request()->auth()` (no matter if the `Authorization` header is set).
- It is **created, but not used** if the site code calls this method, the `Authorization` header is not set when the cache is created but *is* set when the cache is used.
- It is **not created at all** if the site code calls the method and the `Authorization` header is set when the cache would be created.

@bnomei In our tests with the Starterkit and the latest 3.7.0-RC.3, we couldn't reproduce this issue. This suggests that your site code or any plugin may be calling `$kirby->request()->auth()`. Could you please check if that's the case?

**Comment by bnomei** [comment]
thanks for explaining so thoroughly. i will start by removing plugins and report back.

**Comment by bnomei** [comment]
i can not by my best efforts replicate the issue again. even with the same or other projects. something was wrong with my setup (valet, nginx) that day if a simple reboot fixed it. i will close the issue but open it again if i ever happen to run into it again.
